### PR TITLE
Hot fix: remove warning if off by 1 due to odd number of reference times

### PR DESCRIPTION
**Pull request by kaitejohnson**
## Description

This PR fixes the current warning messaging in `allocate_reference_times()`, so that if `prop_delay != prop_delay_used` the user is told this could be due to minimum requirements but could also be due to rounding. Added a test for the rounding case, both when it should and shouldn't warn.

Scope creep:
- add `.Rdata` to `.gitignore`

I think we might want to consider moving this from a warning to a message. Thoughts?

## Checklist

- [ ] My PR is based on a package issue and I have explicitly linked it.
- [ ] I have included the target issue or issues in the PR title in the for Issue(s) *issue-numbers*: PR title
- [X] I have read the [contribution guidelines](https://github.com/epinowcast/.github/blob/main/CONTRIBUTING.md).
- [X] I have tested my changes locally.
- [X] I have added or updated unit tests where necessary.
- [ ] I have updated the documentation if required.
- [X] My code follows the established coding standards.
- [ ] I have added a news item linked to this PR.
- [X] I have reviewed CI checks for this PR and addressed them as far as I am able.

## Summary by CodeRabbit

- New Features
  - Clearer warning when reallocation occurs due to minimum requirements or rounding.
- Bug Fixes
  - Improved allocation logic to account for rounding when matching target delay proportions, reducing false mismatches and ensuring consistent results in edge cases.
- Tests
  - Added test coverage for rounding scenarios, including cases with and without warnings, validating resulting counts.
- Chores
  - Updated ignore rules to exclude RData files and adjusted tracked directories.

### CODE REVIEW

File: R/allocate_reference_times.R [hunk 1]
```diff
@@ -287,10 +287,10 @@ allocate_reference_times <- function(reporting_triangle,
 
   prop_delay_used <- n_history_delay / n_used
 
-  if (prop_delay_used != prop_delay) {
+  if (round(prop_delay_used, 3) != round(prop_delay, 3)) {
```
Comment (seabbs) [inline R/allocate_reference_times.R:290]: Why can't it tell us which it is?

**Review by seabbs — APPROVED** [review]

**Comment by kaitejohnson** [comment]
Tried to test with regexps but testthat apparently can't handle just look for the second part of a message...
